internal/utils.go

```diff
@@ -0,0 +1,72 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/codecrafters-io/tester-utils/logger"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+type GitTempDir struct {
```

Comment:
Logger is not being used, `TempDir` can be inferred from `TempGitPath`. I'd remove this entirely and just call `restoreGit(newPath string, originalPath string)` directly. Not worth extracting out